### PR TITLE
Loosen doctrine/common requirement

### PR DESCRIPTION
**Pull request by bburnichon**
Also update dev-master branch-alias to be in line with master branch

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
     ],
     "require": {
         "php": ">=5.3.2",
-        "doctrine/common": ">=2.4,<2.6-dev"
+        "doctrine/common": "~2.4"
```
Comment (deeky666) [inline composer.json]: Not sure about this one. /cc @Ocramius

Comment (Ocramius) [inline composer.json]: We should probably move it to `<2.7-dev`

File: composer.json [hunk 2]
```diff
@@ -28,7 +28,7 @@
     },
     "extra": {
         "branch-alias": {
-            "dev-master": "2.5.x-dev"
+            "dev-master": "2.6.x-dev"
```
Comment (GrahamCampbell) [inline composer.json]: :-1:

Comment (GrahamCampbell) [inline composer.json]: This change is both unrelated, and unneeded. Composer only respects the alias on the actual master branch.

**Comment by Ocramius** [comment]
Moved to #2268

**Comment by mickaelandrieu** [comment]
To be honest, this put your package with requirements >= PHP 5.4 cause doctrine/common 2.6.x requirements are php >= 5.5.

This force us (PrestaShop team) to explicitely declare the version of doctrine/common we want to require :(

**Comment by Ocramius** [comment]
> This force us (PrestaShop team) to explicitely declare the version of doctrine/common we want to require :(

Uhhh... no? Composer will resolve to `doctrine/common:2.5.x` during `composer update`

**Comment by mickaelandrieu** [comment]
> Uhhh... no? Composer will resolve to doctrine/common:2.5.x during composer update

Nope, with "doctrine/dbal": "~2.5.3" we get "doctrine/common": "2.6.1".

**Comment by Ocramius** [comment]
@mickaelandrieu that depends on the version of PHP that you have installed?

**Comment by mickaelandrieu** [comment]
@Ocramius weird, Composer does'nt works as expected on this point :/ We got a developper from our team able to install doctrine/common 2.6 with PHP 5.4 installed .. 

But it's not related to your package, it's probably an issue of composer :/

Sorry for disturbing and thank you for your time :)

**Comment by Ocramius** [comment]
@mickaelandrieu check CLI vs WEB envs

**Comment by mickaelandrieu** [comment]
Ok got it <3 

We just forgot to update dependencies after switching php versions :lollipop: 

1 / PHP 5.6 => launch tests OK
2 / PHP 5.5 => launch tests OK
3 / PHP 5.4 => launch tests KABOOM 

Thank you for help :+1:

**Comment by Ocramius** [comment]
At least it is fixed now :-)

**Pull request closed without merging**